datasets/Fejza_2023/compose.py

```diff
@@ -0,0 +1,161 @@
+import pandas as pd
+import sys
+import re
+from urllib.request import urlopen
+
+sys.path.append("../../scripts")
+import utils
+
+# Read input
+wos1 = pd.read_excel(
+    "https://zenodo.org/records/8348665/files/ECM&ICIs&marker_wos.xls?download=1"
+)
+wos2 = pd.read_excel(
+    "https://zenodo.org/records/8348665/files/ECM&ICIs_wos.xls?download=1"
+)
+wos3 = pd.read_excel(
+    "https://zenodo.org/records/8348665/files/matrixRemodeling&ICIS_wos.xls?download=1"
+)
+wos4 = pd.read_excel(
+    "https://zenodo.org/records/8348665/files/matrixRemodeling&immunoth_wos.xls?download=1"
+)
+wos = pd.concat([wos1, wos2, wos3, wos4])
+wos = utils.extract_doi(wos, "DOI")
+wos = utils.extract_pmid(wos, "Pubmed Id")
+wos = utils.rename_columns(wos, title="Article Title", year="Publication Year")
+
+sco1 = pd.read_csv(
+    "https://zenodo.org/records/8348665/files/ECM&ICIs_scopus.csv?download=1"
+)
+sco3 = pd.read_csv(
+    "https://zenodo.org/records/8348665/files/ECM&immunother_scopus.csv?download=1"
+)
+sco4 = pd.read_csv(
+    "https://zenodo.org/records/8348665/files/matrixRemodeling&ICIs_scopus.csv?download=1"
+)
+sco5 = pd.read_csv(
+    "https://zenodo.org/records/8348665/files/matrixRemodeling&immunother_scopus.csv?download=1"
+)
+
+sco = pd.concat([sco1, sco3, sco4, sco5])
+sco = utils.extract_doi(sco, "DOI")
+sco = utils.rename_columns(sco, title="Title", year="Year")
+
+sco2 = pd.read_csv(
+    "https://zenodo.org/records/8348665/files/ECM&immunoth&marker_scopus.csv?download=1",
+    sep="\t",
+    encoding = "ISO-8859-1"
+)
+sco2 = utils.extract_doi(sco2, "DOI", "", "&")
+sco2 = utils.extract_doi(sco2, "DOI.1", "", "&")
+sco2 = utils.extract_pmid(sco2, "PMID")
+for i in range(25):
+    if i not in [8, 11, 12]:
+        column_name = "Unnamed: " + str(i + 15)
+        sco2 = utils.extract_doi(sco2, column_name, "", "&")
+
+pubs = [
+    "https://zenodo.org/records/8348665/files/summary-ECM&ICI_pubmed.txt?download=1",
+    "https://zenodo.org/records/8348665/files/summary-ECM&immunother_pubmed.txt?download=1",
+    "https://zenodo.org/records/8348665/files/summary-ECMremodeling&ICI_pubmed.txt?download=1",
+    "https://zenodo.org/records/8348665/files/summary-ICI&ECM&marker_pubmed.txt?download=1",
+    "https://zenodo.org/records/8348665/files/summary-matrix_remodeling&immunother_pubmed.txt?download=1",
+]
+
+pmids = []
+
+# files are in pubmed txt file, we aim to just extract the pmid
+for filename in pubs:
+    file = urlopen(filename)
+    current_line = ""
+    current_nr = 1
+    # loop over lines and concatenate separate entries to 1 line so we can extract from it
+    for line in file:
+        line = str(line)
+        line = line.replace("b'", "")
+        line = line.replace('b"', "")
+        line = line.replace("\\r\\n'", " ")
```

Comment:
```suggestion
         line = line.decode("utf-8").strip()
```
I think we should use the decode function here

Reply:
Decoding with utf-8 failed sadly enough :(. I could look into it further, but not important enough I think.

Reply:
Interesting, this worked locally for me: `line = line.decode("utf-8").strip()`

Reply:
hmm, okay. I did only try decoding the entire file at once, not 1 line at the time. So will implement this :).

Reply:
interesting, good to know! Thanks

Reply:
Ok, the decoding works, but gives incorrect results due to how I handle it afterwards. (spacing issue I think with the regex match). So I did keep the original in.

Reply:
Haha that's probably thanks to the .strip(), if a line break is right between "PMID:" and the id itself, nice.
I think I would still prefer to change the regex rather than do byte to string conversion manually, for maintainability.

for example: `re.search(r"PMID:\s*(\d+)", current_entry_text)`

If you don't agree, I would at least want explicit comments explaining how and why we do manual byte decoding.

Reply:
Yes it was. But indeed we can just adjust the regex like you suggested. Thanks. (adjusted the PR)